pandas/tests/frame/test_analytics.py

```diff
@@ -23,54 +23,192 @@
 
 import pandas.util.testing as tm
 import pandas.util._test_decorators as td
-from pandas.tests.frame.common import TestData
 
 
-class TestDataFrameAnalytics(TestData):
+def assert_stat_op_calc(opname, alternative, main_frame, has_skipna=True,
```

Comment:
is there a reason you split up these to calc/api tests now? its very hard to tell if anything actually changed. I am concerned that something DID change even accidently. can you just do this as a copy-paste change of the original function.

Reply:
Yes, because they perform different checks, need different fixtures, and introduced the redundancy mentioned further up the thread (of checking the api-part twice in e.g. `test_max`).

I did not remove any single line of the functions.

> can you just do this as a copy-paste change of the original function.

you mean as a separate commit?

Reply:
> no i mean do a pre-cursor PR that moves the functions, but doesn't change them. Then a followon to show those changes. moving & changing need to be separate steps. It should be easy to see what actually changed.

Why not a separate commit...? You can review commits individually. Plus, this is essentially linked with the whole fixturization effort, which would have to come first (and your request to make `_check_stat_op` a module-level function is strictly speaking orthogonal to this PR).

I'll make some nice modular commits for you to review in sequence.

Reply:
you can try, but its much easier to review a separate PR to be honest

Reply:
I pushed some commits that just go until moving `_check_stat_op` and `_check_bool_op` and making them run. I'll push the rest of the commits once the CI passes.